src/Adapter/Presenter/Cart/CartPresenter.php

```diff
@@ -505,6 +505,16 @@ private function getTemplateVarVouchers(Cart $cart)
                 $cartVoucher['reduction_formatted'] = $cartVoucher['reduction_percent'] . '%';
             } elseif (isset($cartVoucher['reduction_amount']) && $cartVoucher['reduction_amount'] > 0) {
                 $value = $this->includeTaxes() ? $cartVoucher['reduction_amount'] : $cartVoucher['value_tax_exc'];
+                $currencyFrom = new \Currency($cartVoucher['reduction_currency']);
+                $currencyTo = new \Currency($cart->id_currency);
+                if ($currencyFrom->conversion_rate == 0) {
+                    $value = 0;
+                } else {
+                    // convert to default currency
+                    $value /= $currencyFrom->conversion_rate;
```

Comment:
I'm worried about rounding issues when I see this 😐 . Am I too paranoid ?

Reply:
You're not

Reply:
unfortunately I've no tool to handle it correctly, nor guidelines to know how to round it.
following call $this->priceFormatter->convertAndFormat($value); will round correctly if the precision is large enough, which will be the case if the conversion rate is not 1000000+ ^^